### ISSUE TITLE
[FEA] implement series.shift series.rank and series.transfrom

### ISSUE THREAD

**Issue by AK-ayush** [rapidsai/cudf#1232]
It is very useful to have following operations in series:
* [`Series.shift()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.shift.html#pandas-series-shift)
* [`Series.rank()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.rank.html#pandas-series-rank)
* [`Series.transform()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.transform.html#pandas-series-transform)
* [`Series.rolling()` ](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.rolling.html#pandas-series-rolling)

And few others such as [`cum_sum()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.core.groupby.DataFrameGroupBy.cumsum.html#pandas-core-groupby-dataframegroupby-cumsum), [`cum_count()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.core.groupby.GroupBy.cumcount.html#pandas-core-groupby-groupby-cumcount) and etc. within `groupby` would also be very useful in order to migrate from existing pandas code to cudf. 

If some of these already existing in cudf please let me know i'll update it.
cc @randerzander

**Comment by beckernick** [comment]
@AK-ayush I'm going to separate these out into separate feature requests, as:

1) These can be worked on in parallel
2) Some of these capabilities are likely more important than others

When I do that, I'll close this ticket in favor of the new ones.

**Comment by AK-ayush** [comment]
Sounds good :)

**Comment by beckernick** [comment]
@AK-ayush is the specific `transform` API desirable or the functionality itself.

For example, is the combination of `applymap` for Series and `apply_rows`/`apply_chunks` for DataFrames clean enough for UDFs? At the Series level, you can do the examples from `pd.Series.transform` fairly well.

```
import cudf

ds = cudf.Series(range(3)).astype('float32')
print(ds)

def udf(x):
    return x**.5

print(ds.applymap(udf))
0    0.0
1    1.0
2    2.0
dtype: float32
0          0.0
1          1.0
2    1.4142135
dtype: float32

```

**Comment by AK-ayush** [comment]
@beckernick I guess `applymap` on series will be enough for my use case. 
Also, I'll be happy to help in any of the above features, only thing is that, I know CPU implementation but not sure how to use reduction for GPU implementation.

**Comment by beckernick** [comment]
Sounds good. I've split these up into separate tickets (except for transform, given your comment above), and will close this issue now.
